### PR TITLE
Added initial ARM template

### PR DESCRIPTION
**Pull request by dalemorson**
This is the initial ARM template to deploy the following resources:

- App Service Plan
- App Service Certificate
- App Service
- App Insights

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -0,0 +1,211 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "environmentName": {
```
Comment (EwanNoble) [inline azure/template.json]: This requires two different values as resource naming differs from runtime environment name

File: azure/template.json [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+                                "value": "[reference('application-insights').outputs.InstrumentationKey.value]"
+                            },
+                            {
+                                "name": "ASPNETCORE_ENVIRONMENT",
+                                "value": "[toUpper(parameters('environmentName'))]"
+                            },
+                            {
+                                "name": "WEBSITE_LOAD_CERTIFICATES",
+                                "value": "[reference('app-service-certificate').outputs.certificateThumbprint.value]"
+                            }
```
Comment (EwanNoble) [inline azure/template.json]: The app settings need the configuration connection string

File: azure/template.json [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+                    "aspInstances": {
+                        "value": "[parameters('aspInstances')]"
+                    }
+                }
+            }
+        },
+        {
+            "apiVersion": "2017-05-10",
+            "name": "app-service-certificate",
+            "resourceGroup": "[parameters('sharedResourceGroup')]",
```
Comment (EwanNoble) [inline azure/template.json]: Why is this certificate going to the shared resource group?

File: azure/template.json (hunk 1 above)
Comment (dalemorson) [reply azure/template.json]: I've updated the template to use environmentFullName and resourceEnvironmentName parameters correctly.

File: azure/template.json (hunk 2 above)
Comment (dalemorson) [reply azure/template.json]: I've added in the configurationStorageConnectionString parameter and added this into the appServiceAppSettings parameters for the App Service.

File: azure/template.json (hunk 3 above)
Comment (dalemorson) [reply azure/template.json]: Removed the app-service-certificate deployment from being deployed to a shared resource group as there is a new app service plan being deployed specifically for this app service. If the app service were being deployed to a shared app service plan, then I could of used this method.

**Pull request merged**